tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java

```diff
@@ -441,36 +472,52 @@ private void printMembers(Map<String, Entry<Optional<GroupState>, Optional<Colle
+                            String upgraded = memberAssignment.upgraded.map(Object::toString).orElse(MISSING_COLUMN_VALUE);
+                            System.out.printf(verboseFormat, currentEpoch, currentAssignment, targetEpoch, targetAssignment, upgraded);
                         }
                         System.out.println();
                     }
                 }
             });
         }
 
-        private void printStates(Map<String, GroupInformation> states) {
+        private String getAssignmentString(List<TopicPartition> assignment) {
+            Map<String, List<TopicPartition>> grouped = new HashMap<>();
+            assignment.forEach(
+                tp -> grouped.computeIfAbsent(tp.topic(), key -> new ArrayList<>()).add(tp));
+            return grouped.entrySet().stream().map(entry -> {
+                String topicName = entry.getKey();
+                List<TopicPartition> topicPartitions = entry.getValue();
+                return topicPartitions.stream().map(TopicPartition::partition).map(Object::toString).sorted().collect(Collectors.joining(",", topicName + ":", ""));
+            }).sorted().collect(Collectors.joining(";"));
+        }
+
+        private void printStates(Map<String, GroupInformation> states, boolean verbose) {
             states.forEach((groupId, state) -> {
                 if (shouldPrintMemberState(groupId, Optional.of(state.groupState), Optional.of(1))) {
                     String coordinator = state.coordinator.host() + ":" + state.coordinator.port() + "  (" + state.coordinator.idString() + ")";
                     int coordinatorColLen = Math.max(25, coordinator.length());
+                    int groupColLen = Math.max(15, state.group.length());
 
-                    String format = "\n%" + -coordinatorColLen + "s %-25s %-20s %-15s %s";
+                    String assignmentStrategy = state.assignmentStrategy.isEmpty() ? MISSING_COLUMN_VALUE : state.assignmentStrategy;
 
-                    System.out.printf(format, "GROUP", "COORDINATOR (ID)", "ASSIGNMENT-STRATEGY", "STATE", "#MEMBERS");
-                    System.out.printf(format, state.group, coordinator, state.assignmentStrategy, state.groupState, state.numMembers);
+                    if (verbose) {
+                        String format = "\n%" + -groupColLen + "s %" + -coordinatorColLen + "s %-20s %-20s %-15s %-25s %s";
+                        String groupEpoch = state.groupEpoch.map(Object::toString).orElse(MISSING_COLUMN_VALUE);
+                        String targetAssignmentEpoch = state.targetAssignmentEpoch.map(Object::toString).orElse(MISSING_COLUMN_VALUE);
+                        System.out.printf(format, "GROUP", "COORDINATOR (ID)", "ASSIGNMENT-STRATEGY", "STATE", "GROUP-EPOCH", "TARGET-ASSIGNMENT-EPOCH", "#MEMBERS");
+                        System.out.printf(format, state.group, coordinator, assignmentStrategy, state.groupState, groupEpoch, targetAssignmentEpoch, state.numMembers);
```

Comment:
nit: Should we just inline groupEpoch and targetAssignmentEpoch like you did in the other cases?